### PR TITLE
Added retain parameter to MQTT publish action

### PR DESCRIPTION
**Pull request by dakhnod**
Makes it possible to decide publishing a message retained or not per situation,
not only through the binding config.

### CODE REVIEW

File: bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/action/MQTTActions.java [hunk 1]
```diff
@@ -76,6 +76,37 @@ public void publishMQTT(
         });
     }
 
+    @RuleAction(label = "@text/actionLabel", description = "@text/actionDesc")
+    public void publishMQTT(
+            @ActionInput(name = "topic", label = "@text/actionInputTopicLabel", description = "@text/actionInputTopicDesc") @Nullable String topic,
+            @ActionInput(name = "value", label = "@text/actionInputValueLabel", description = "@text/actionInputValueDesc") @Nullable String value,
+            @ActionInput(name = "retain", label = "@text/actionInputRetainlabel", description = "@text/actionInputRetainDesc") @Nullable boolean retain) {
```
Comment (davidgraeff) [inline bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/action/MQTTActions.java]: Use `Boolean` if you want this parameter to be nullable. Language types like `boolean` have a defined state at every time (either true, false in this case).

**Comment by davidgraeff** [comment]
Could you please make the 2 argument method use your new 3 argument method? There's code that can be shared.

File: bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/action/MQTTActions.java (hunk 1 above)
Comment (dakhnod) [reply bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/action/MQTTActions.java]: oh, you are right

**Comment by dakhnod** [comment]
> Could you please make the 2 argument method use your new 3 argument method? There's code that can be shared.

voila

**Comment by dakhnod** [comment]
> Please also add the static method to make it available in Ruesl DSL.

oui

**Comment by J-N-K** [comment]
Thanks. Before we can merge, please sign-off your commits. If you don't know how to do so, please click on "Details" in the "DCO" line above.

**Comment by dakhnod** [comment]
wtf is happening...i am reopening this pull request since github is for some reason killing me due to atom and german special chars etc...

**Comment by davidgraeff** [comment]
never use git merge and you don't have that problem :D

**Pull request closed without merging**